File: BackEndAprovacao/BackEndAprovacao/Models/Processo.cs
using BackEndAprovacao.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace BackEndAprovacao.Models
{
    public class Processo : EntidadeBase
    {
        [Required]
        [StringLength(20, MinimumLength = 20, ErrorMessage = "Verifique se o numero do processo foi digitado corretamente")]
        public string NumeroDeProcesso { get; set; }
        public double ValorCausa { get; set; }
        public int ReclamanteId { get; set; }
        public virtual Reclamante Reclamante { get; set; }
        public int EscritorioId { get; set; }
        public virtual Escritorio Escritorio { get; set; }
        public EProcessoEstado EstadoId { get; set; }
        public string Estado => EstadoId.ObterDescricao();
        public bool Ativo { get; set; }

        public Processo()
        {
            EstadoId = EProcessoEstado.Pendente;
        }
    }
}

File: front-end/src/app/models/interface/Processo.ts
import { EProcessoEstado } from './EProcessoEstado.enum';
import { Escritorio } from './Escritorio';
import { Reclamante } from './Reclamante';
export interface Processo {
  id?: number;
  numeroDeProcesso: string;
  valorCausa: number;
  reclamanteId: number;
  escritorioId: number;
  reclamante?: Reclamante;
  escritorio?: Escritorio;
  estadoId?:EProcessoEstado;
  estado?:string;
  ativo:boolean;
}
